controlplane/src/core/bufservices/PlatformService.ts

```diff
@@ -1649,47 +2109,48 @@ export default function (opts: RouterOptions): Partial<ServiceImpl<typeof Platfo
         for (const composition of result.compositions) {
           await schemaCheckRepo.createCheckedFederatedGraph(schemaCheckID, composition.id, limit);
 
-          // We collect composition errors for all federated graphs
-          if (composition.errors.length > 0) {
-            for (const error of composition.errors) {
-              compositionErrors.push({
-                message: error.message,
-                federatedGraphName: composition.name,
-                namespace: composition.namespace,
-              });
-            }
+          for (const error of composition.errors) {
+            compositionErrors.push({
+              message: error.message,
+              federatedGraphName: composition.name,
+              namespace: composition.namespace,
+            });
           }
 
           // We don't collect operation usage when we have composition errors or
           // when we don't have any inspectable changes. That means any breaking change is really breaking
-          if (composition.errors.length === 0 && isInspectable && inspectorChanges.length > 0) {
-            if (limit <= 0) {
-              continue;
-            }
+          if (composition.errors.length > 0 || !isInspectable || inspectorChanges.length === 0) {
+            continue;
+          }
 
-            const result = await trafficInspector.inspect(inspectorChanges, {
-              daysToConsider: limit,
-              federatedGraphId: composition.id,
-              organizationId: authContext.organizationId,
-            });
+          if (limit <= 0) {
```

Comment:
Do we really need it here? If not, remove.

Reply:
Removed